src/runner.py

```diff
@@ -548,6 +553,83 @@ def _should_render_templates_with_metrics(self) -> bool:
         """
         return self._shared_fs is not None
 
+    def _configure_aproxy(self, aproxy_address: str) -> None:
+        """Configure aproxy.
+
+        Args:
+            aproxy_address: Proxy to configure aproxy with.
+
+        Raises:
+            RunnerAproxyError: If unable to configure aproxy.
+        """
+        if self.instance is None:
+            raise RunnerError("Runner operation called prior to runner creation.")
+
+        logger.info("Configuring aproxy for the runner.")
+
+        self.instance.execute(["snap", "set", "aproxy", f"proxy={aproxy_address}"])
+
+        stdout, _ = execute_command(
+            [
+                "/usr/bin/awk",
+                f'$4 == "{self.instance.name}" {{ print $3 }}',
+                LXDBR_DNSMASQ_LEASES_FILE,
+            ],
+            check_exit=True,
+        )
```

Comment:
When using `awk`, if the name doesn't exist in the file, `awk` will print an empty line and `exit 0`. Maybe we should check the return value here, or using Python to parse the file and raise an exception.